### PR TITLE
Use one experiment per optimization

### LINKED ISSUE #118: Revisit mapping from Everest optimization to ERT experiments
Currently we spawn multiple experiments, each with one ensemble. If we are to use ERT storage to store optimization results, this hierarchy should be the most sensible for what we need. Most likely it should be one experiment per optimization, and one ensemble per batch.

Proposed solution: In `simulator.py`, create one experiment named by "case", then create ensembles in `BatchSimulator.start()`.

### PR DESCRIPTION
**Pull request by yngve-sk**
and one ensemble per batch.

Closes #118

### CODE REVIEW

File: src/everest/export.py [hunk 1]
```diff
@@ -162,7 +162,9 @@ def get_internalized_keys(config: EverestConfig, batch_ids: Optional[Set[int]] =
     with open_storage(config.storage_dir, "r") as storage:
         for batch_id in batch_ids:
             case_name = f"batch_{batch_id}"
-            experiment = storage.get_experiment_by_name(f"experiment_{case_name}")
+            experiments = [*storage.experiments]
+            assert len(experiments)
+            [experiment] = experiments
```
Comment (verveerpj) [inline src/everest/export.py:167]: I do not find this very clear code, why not just `experiment = next(storage.experiments)`? An empty `storage.experiments` would raise an exception, which should indicate a bug.

Also the assert should probably be `== 1`?

File: src/everest/export.py [hunk 2]
```diff
@@ -313,7 +315,11 @@ def _load_simulation_data(
         # pylint: disable=unnecessary-lambda-assignment
         def load_batch_by_id():
             case_name = f"batch_{batch}"
-            experiment = storage.get_experiment_by_name(f"experiment_{case_name}")
+            experiments = [*storage.experiments]
+
+            # Always assume 1 experiment per simulation/enspath, never multiple
+            assert len(experiments) == 1
+            [experiment] = experiments
```
Comment (verveerpj) [inline src/everest/export.py:322]: As above.

File: src/everest/simulator/simulator.py [hunk 3]
```diff
@@ -132,7 +135,18 @@ def __call__(
...
+            if self._experiment_id is None:
+                experiment = storage.create_experiment(
+                    name=f"EnOpt@{datetime.now().strftime('%Y-%m-%d@%H:%M:%S')}",
+                    parameters=self.ert_config.ensemble_config.parameter_configuration,
+                    responses=self.ert_config.ensemble_config.response_configuration,
+                )
+
+                self._experiment_id = experiment.id
+
+            experiment = storage.get_experiment(self._experiment_id)
```
Comment (verveerpj) [inline src/everest/simulator/simulator.py:147]: Could be in an else clause, right? Any reason to pull the experiment again out of the storage if we just made it?

File: tests/everest/test_everest_output.py [hunk 4]
```diff
@@ -19,6 +20,28 @@
 from tests.everest.utils import relpath, tmpdir
 
 
+@pytest.mark.redundant_test  # superseded by test_math_func_advanced and _multiobj
```
Comment (verveerpj) [inline tests/everest/test_everest_output.py:23]: Probably should not be a redundant test?

File: tests/everest/test_everest_output.py [hunk 5]
```diff
@@ -19,6 +20,28 @@
...
+    workflow.start_optimization()
+
+    batches = os.listdir(config.simulation_dir)
+    ert_config = ErtConfig.with_plugins().from_dict(everest_to_ert_config(config))
+    enspath = ert_config.ens_path
+
+    with open_storage(enspath, mode="r") as storage:
+        experiments = [*storage.experiments]
+        assert len(experiments) == 1
+        [experiment] = experiments
```
Comment (verveerpj) [inline tests/everest/test_everest_output.py:39]: As above, although checking that there is only one experiment is probably good.

**Review by verveerpj — APPROVED** [review]
LGTM!

I have some small comments, but they are not fundamental issues, just questions.

File: src/everest/simulator/simulator.py (hunk 3 above)
Comment (yngve-sk) [reply src/everest/simulator/simulator.py:147]: Not really, putting it in an else clause

**Comment by oyvindeide** [comment]
Closing as development is continuing in ert

**Pull request closed without merging**